app/models/project_services/composer_service.rb

```diff
@@ -0,0 +1,475 @@
+          ['Symfony1 Plugin', 'symfony1-plugin'],
+          ['Tusk Asset', 'tusk-asset'],
+          ['Tusk Command', 'tusk-command'],
+          ['Tusk Task', 'tusk-task'],
+          ['TYPO3 CMS Extension', 'typo3-cms-extension'],
+          ['TYPO3 Flow Boilerplate', 'typo3-flow-boilerplate'],
+          ['TYPO3 Flow Build', 'typo3-flow-build'],
+          ['TYPO3 Flow Framework', 'typo3-flow-framework'],
+          ['TYPO3 Flow Package', 'typo3-flow-package'],
+          ['TYPO3 Flow Plugin', 'typo3-flow-plugin'],
+          ['TYPO3 Flow Site', 'typo3-flow-site'],
+          ['Wolf CMS Plugin', 'wolfcms-plugin'],
+          ['WordPress Core', 'wordpress-core'],
+          ['WordPress Must Use Plugin', 'wordpress-muplugin'],
+          ['WordPress Plugin', 'wordpress-plugin'],
+          ['WordPress Theme', 'wordpress-theme'],
+          ['Zend Extra', 'zend-extra'],
+          ['Zend Library', 'zend-library'],
+          ['Zend Module', 'zend-module'],
+          ['Zikula Module', 'zikula-module'],
+          ['Zikula Theme', 'zikula-theme']
+        ],
+        default_choice: 'library'
+      },
+      { type: 'fieldset', legend: 'Branches:', fields:
+        [
+          { type: 'checkbox',
+            name: 'export_branches',
+            label: 'Branch Listing'
+          },
+          { type: 'text',
+            name: 'branch_filters',
+            label: 'Branch Filters',
+            placeholder: 'branches you wish to export comma seperated.',
+            hint: 'Separate branches with commas. Leave blank to export all branches.'
+          }
+        ]
+      },
+      { type: 'fieldset', legend: 'Tags:', fields:
+        [
+          { type: 'checkbox',
+            name: 'export_tags',
+            label: 'Tag Listing'
+          },
+          { type: 'text',
+            name: 'tag_filters',
+            label: 'Tag Filters',
+            placeholder: 'tags you wish to export comma seperated.',
+            hint: 'Separate tags with commas. Leave blank to export all tags.'
+          }
+        ]
+      },
+      { type: 'fieldset', legend: 'Advanced:', fields:
+        [
+          { type: 'textarea',
+            name: 'custom_json',
+            label: 'Custom JSON',
+            placeholder: 'custom composer.json to use for exporting this package.'
+          }
+        ]
+      }
+    ]
+  end
+
+  #disable test button
+  def can_test?
+    false
+  end
+
+  def process_packages_on_save
+
+      #process packages for all tags
+      project.repository.tags.each do |tag|
+        process_commit(tag)
+      end
+
+      # process packages for all branches
+      project.repository.branches.each do |branch|
+        process_commit(branch)
+      end
+
+  end
+
+  def process_commit(ref)
+
+    previous_package_removed = false
+    begin
+      if commit_was_activated? && commit_was_exported?(ref)
+        defaults = package_mode_was == 'advanced' ? ActiveSupport::JSON.decode(custom_json_was) : {'type'=>package_type_was}
```

Comment:
Line is too long. [124/80]<br>Surrounding space missing for operator '=>'.<br>Space inside { missing.<br>Space inside } missing.